src/query/functions/src/scalars/map.rs

```diff
@@ -229,6 +241,125 @@ pub fn register(registry: &mut FunctionRegistry) {
+            DataType::Map(box DataType::Tuple(type_tuple)) => {
+                if type_tuple.len() == 2 {
+                    (type_tuple[0].clone(), type_tuple[1].clone())
+                } else {
+                    return None;
+                }
+            }
+            DataType::EmptyMap => (DataType::EmptyMap, DataType::EmptyMap),
+            _ => {
+                return None;
+            }
+        };
+
+        let mut in_key_types: Option<DataType> = None;
+
+        for arg_type in args_type.iter().skip(1) {
+            match arg_type {
+                DataType::Array(box key_type) => {
+                    if map_key_type == DataType::EmptyMap {
+                        if let Some(in_key_type) = &in_key_types {
+                            if in_key_type != key_type {
+                                return None;
+                            }
+                        } else {
+                            in_key_types = Some(key_type.clone());
+                        }
+                    } else if *key_type != map_key_type {
+                        return None;
+                    }
+                }
+                DataType::Null => {
+                    return None;
+                }
+                _ => {
+                    return None;
+                }
+            }
+        }
+
+        let return_type = args_type[0].clone();
+
+        Some(Arc::new(Function {
+            signature: FunctionSignature {
+                name: "map_delete".to_string(),
+                args_type: args_type.to_vec(),
+                return_type: return_type.clone(),
+            },
+            eval: FunctionEval::Scalar {
+                calc_domain: Box::new(|_, _| FunctionDomain::Full),
+                eval: Box::new(move |args, _ctx| {
+                    let input_length = args.iter().find_map(|arg| match arg {
+                        ValueRef::Column(col) => Some(col.len()),
+                        _ => None,
+                    });
+
+                    let mut output_map_builder =
+                        ColumnBuilder::with_capacity(&return_type, input_length.unwrap_or(1));
+
+                    let delete_key_list = prepare_delete_key_list(args, input_length);
+
+                    for idx in 0..(input_length.unwrap_or(1)) {
+                        let input_map_sref = match &args[0] {
+                            ValueRef::Scalar(map) => map.clone(),
+                            ValueRef::Column(map) => unsafe { map.index_unchecked(idx) },
+                        };
+
+                        let inner_builder_type = match input_map_sref.infer_data_type() {
+                            DataType::Map(box typ) => typ,
+                            DataType::EmptyMap => DataType::EmptyMap,
+                            _ => unreachable!(),
+                        };
+
+                        let mut filtered_kv_builder = ColumnBuilder::with_capacity(
```

Comment:
This builder can be removed, only if the argument is a map need a builder, if the value is `Null` or `EmptyMap`, we can use 
```rust
output_map_builder.push_default();
```

Reply:
Good One, Thankyou